.github/workflows/documentation.yaml

```diff
@@ -1,27 +1,61 @@
-name: Generate terraform docs
+name: generate-terraform-docs
 
 on:
-  pull_request:
+  push:
     branches:
       - main
     paths:
       - '**.tf'
-      - '**.md'
+
 jobs:
-  docs:
+  tf-docs:
+    name: TF docs
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v3
-      with:
-        ref: ${{ github.event.pull_request.head.ref }}
+      - uses: actions/checkout@v3
+        id: actions-checkout
+        with:
+          ref: main
+
+      - id: git-branch
+        run: |
+          generated_branch="update-docs-$(date +%s)"
+          echo "generated_branch=${generated_branch}" >>$GITHUB_OUTPUT
+          git checkout -b $generated_branch
+          git push --set-upstream origin $generated_branch
+
+      - name: Render terraform docs inside the main and the modules README.md files and push changes back to PR branch
+        id: terraform-docs
+        uses: terraform-docs/gh-actions@v1.0.0
+        with:
+          working-dir: .
+          recursive: true
+          recursive-path: modules
```

Comment:
Yes I think so, I need to check the https://github.com/terraform-docs/gh-actions#:~:text=false-,working%2Ddir,-Comma%20separated%20list field

Reply:
Added in https://github.com/equinix-labs/terraform-equinix-kubernetes-addons/pull/27/commits/39201e7f8746baeca949def7e0c9258b309e5ad8

Reply:
WIP, trying to fix https://github.com/equinix-labs/terraform-equinix-kubernetes-addons/actions/runs/3750369213/jobs/6369995788#step:4:38

Reply:
Fixed in https://github.com/equinix-labs/terraform-equinix-kubernetes-addons/pull/27/commits/5d9e98bf5b8f6f266ac6052df68648ee6797e2ea with `find-dir` which discover all tf files using `find ./find\_dir -name \*.tf` command. In case we want to exclude some directories in the future, we will need to create different steps or jobs for each working-dir | recursive-path pairs

Reply:
Do we need a change to the template project to add the documentation template blocks to the examples/simple there?

Reply:
Looks like we'll want to build upon this: https://github.com/equinix-labs/terraform-equinix-template/pull/4/files#diff-6be72b996c93c3a6a69a2ea1380076c96fb384164147280df500c77b22dbbfdb

Reply:
So the `generate_addon_structure.sh` script is overridden some of the files after cloning that template and adding the TF_DOCS delimiters

https://github.com/equinix-labs/terraform-equinix-kubernetes-addons/pull/5/commits/53a9611aa56a2931a97e1beafc069766941fd5ef